lib/client/Client.js

```diff
@@ -126,6 +126,22 @@ export default class Client {
     return this;
   }
 
+  /**
+   * Initializes the client with a list of custom mailbox roles.
+   * Mailboxes having these custom roles will be preserved their role.
+   * 
+   * @param roles {Array} The list of custom roles to be added
+   * 
+   * @return {Client} This Client instance.
+   */
+  withCustomMailboxRoles(roles) {
+    roles.forEach(role => {
+      MailboxRole[role.toUpperCase()] = new MailboxRole(role);
```

Comment:
Generally upon case operation in Java we always pass a locale.

For example take the example of the infamous turkish `i`. A toLowerCase on `I` will not give you the same result on US and turkish locales.

There many more subtile differences.

See http://www.informit.com/articles/article.aspx?p=1751371&seqNum=12 for more details.... There should be some JS equivalent.

Reply:
Unfortunately, there is no JS equivalent.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toUpperCase

Reply:
Héhé so this mean I have one more :trollface: about JS...

Reply:
Well, you didn't read to the end of the page, there's a link to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toLocaleUpperCase

Reply:
Thanks, i to UpperCase with en-US as locale